js/article.js

```diff
@@ -1,49 +1,117 @@
+    this.body = marked(this.body);
+    return template(this);
+  };
 
-  this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
-  this.publishStatus = this.publishedOn ? 'published ' + this.daysAgo + ' days ago' : '(draft)';
+//FUNCTIONALITY TO LOAD ALL ARTICLES IN DECENDING ORDER OF PUBLICATION
+  Article.loadAll = function(inputData) {
+    var articles = inputData.sort(function(a,b) {
+      return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
+    }).map(function(ele) {
+      return new Article(ele);
+    });
+    Article.allArticles = articles;
+  };
 
-  return templateRender(this);
-};
+// THIS ADDS FUNCTIONALITY TO RETRIEVE DATA FROM LOCAL OR REMOTE SOURCE, WHICH PROCESS AND HANDS OFF CONTROL TO THE VIEW ********
+  Article.fetchAll = function(nextFunction) {
+    //if (localStorage.blogArticles) {
+    $.ajax({
+      type: 'HEAD',
+      url: 'data/blogArticles.json',
+      success: function(data, message, xhr) {
+        var eTag = xhr.getResponseHeader('eTag');
+        if (!localStorage.eTag || eTag !== localStorage.eTag) {
+          Article.getAll(nextFunction); // DONE: pass 'nextFunction' into Article.getAll();
+        } else {
+          Article.loadAll(JSON.parse(localStorage.blogArticles));
+          // Replace the following line with 'nextFunction' and invoke.
+          nextFunction();
+        }
+      }
+    });
+  };
 
-//FUNCTIONALITY TO LOAD ALL ARTICLES IN DECENDING ORDER OF PUBLICATION
-Article.loadAll = function(inputData) {
-  inputData.sort(function(a,b) {
-    return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
-  }).forEach(function(ele) {
-    Article.allArticles.push(new Article(ele));
-  });
-};
-
-// THIS ADDS FUNCTIONALITY TO RETRIEVE DATA FROM LOCAL OR REMOTE SOURCE, WHICH PROCESS AND HANDS OFF CONTROL TO THE VIEW
-Article.fetchAll = function() {
-
-  if (localStorage.hackerIpsum) {
-    Article.loadAll(JSON.parse(localStorage.hackerIpsum));
-    articleView.renderIndexPage();
-  } else {
-    $.getJSON('data/hackerIpsum.json', function(data, status, XHR) {
-      localStorage.hackerIpsum = JSON.stringify(data);
-      console.log(data);
-      Article.loadAll(data);
-      articleView.renderIndexPage();
+  Article.getAll = function(nextFunction) {
+    $.getJSON('data/blogArticles.json', function(responseData, message, xhr) {
+      localStorage.eTag = xhr.getResponseHeader('eTag');
+      Article.loadAll(responseData);
+      localStorage.blogArticles = JSON.stringify(responseData);
+      // Invoke our parameter.
+      nextFunction();
     });
   };
-};
+
+  // GETS # OF WORDS IN GIVEN ARTICLE
+  Article.getNumWords = function(article) {
+    return article.body.split(' ').length;
+  };
+
+  Article.numWordsAll = function() {
+    return Article.allArticles.map(function(currentArticle) {
+      return currentArticle.body.match(/\w+/g).length;
+    }).reduce(function(acc, cur) {
+      return acc + cur;
+    });
```

Comment:
Good use of the reduce function. Should be easy to adjust if you want, in order to make it more relevant to your portfolio. I also have it just counting words in my portfolio as well.